build-system/tasks/e2e/mocha-report-multiplexer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const Mocha = require('mocha');
+const {Base} = Mocha.reporters;
+
+/**
+ * Creates a custom Mocha reporter which runs many reporters at once.
```

Comment:
It seems like we now have two ways to combine reporters: This function, and the one below, which combines `MochaDotsReporter` and `JsonReporter`.

https://github.com/ampproject/amphtml/blob/e9cb065f15ef4b2df93775e4e03dd1333e1f5937/build-system/tasks/e2e/mocha-ci-reporter.js#L27-L32

Assuming I'm right, can we consolidate them into one solution (whichever is better)? Okay to punt this to a separate PR, and might be worth getting this bit reviewed by Esther.

Reply:
I looked at `ciReporter` when I was writing this and didn't view it as very extensible. I think it could easily be replaced with an invocation of my new method but as you suggest I would rather handle that as a separate PR. @estherkim Do you have an opinion on this?

Reply:
I would rather have `ciReporter` replaced directly by your junit reporter than in this roundabout way. Ok to do in a separate PR if you want to remove e2e changes out of this one.

Reply:
Alright, I'll move the e2e changes to another PR